### PR TITLE
[tensorexpr] Remove the optional argument in LoopNest::prepareForCodeGen

### PR DESCRIPTION
**Pull request by huiguoo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #66967
* #66452
* __->__ #67144
* #67143
* #66451
* #66450

Differential Revision: [D31881150](https://our.internmc.facebook.com/intern/diff/D31881150)

### CODE REVIEW

**Comment by huiguoo** [comment]
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31881150).

File: torch/csrc/jit/tensorexpr/kernel.cpp [hunk 1]
```diff
@@ -767,7 +767,7 @@ StmtPtr TensorExprKernel::transformLoops(BackendType backendType, StmtPtr st) {
   if (pre_alloc_) {
     auto interm_bufs = l.getIntermediateBufs();
     preAllocIntermediateBufs(interm_bufs);
-    l.prepareForCodegen(interm_bufs);
+    l.prepareForCodegen();
```
Comment (ZolotukhinM) [inline torch/csrc/jit/tensorexpr/kernel.cpp:770]: Please move `l.prepareForCodegen()` out from the if and remove the `else` branch.

**Review by ZolotukhinM — APPROVED** [review]
Looks good!

**Comment by huiguoo** [comment]
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31881150).

**Comment by huiguoo** [comment]
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31881150).

**Comment by huiguoo** [comment]
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31881150).

**Comment by huiguoo** [comment]
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31881150).

**Comment by huiguoo** [comment]
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31881150).

**Comment by huiguoo** [comment]
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31881150).

**Comment by huiguoo** [comment]
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31881150).

**Comment by huiguoo** [comment]
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31881150).

**Pull request closed without merging**